src/force_latest_compatible_version.jl

```diff
@@ -0,0 +1,75 @@
+function _force_latest_compatible_version(ctx::Types.Context; target_name = nothing)
+    direct_deps = load_direct_deps(ctx.env)
+    direct_deps_uuids = getproperty.(direct_deps, Ref(:uuid))
+    nonstdlib_direct_deps_uuids = filter(x -> x ∉ keys(stdlibs()), direct_deps_uuids)
+    return _force_latest_compatible_version(ctx, nonstdlib_direct_deps_uuids; target_name)
+end
+
+function _force_latest_compatible_version(ctx::Types.Context, uuid_list::AbstractVector{Base.UUID};
+                                          target_name = nothing)
+    isempty(uuid_list) && return true
+    results = _force_latest_compatible_version.(Ref(ctx), uuid_list; target_name)
+    return all(results)
+end
+
+function _warn_no_compat(has_compat::Bool,
+                         target_name::AbstractString,
+                         name::AbstractString,
+                         uuid::Base.UUID,
+                         active_version::Base.VersionNumber)
+    if !has_compat
+        if name != target_name
+            @warn(
+                "Package does not have a [compat] entry",
+                name, uuid, active_version, target_name,
+            )
+        end
+    end
+    return nothing
+end
+
+function _force_latest_compatible_version(ctx::Types.Context, uuid::Base.UUID;
+                                          target_name= nothing)
+    manifest_deps = load_manifest_deps(ctx.env.manifest)
+    manifest_deps_uuids = getproperty.(manifest_deps, Ref(:uuid))
+    idx = only(findall(manifest_deps_uuids .== uuid))
+    dep = manifest_deps[idx]
```

Comment:
Can't you just do `ctx.env.manifest[uuid]` and use the `PackageEntry` it returns?

Reply:
Done.